awscrt/mqtt.py

```diff
@@ -21,10 +21,29 @@
 
 
 class QoS(IntEnum):
-    """Quality of Service"""
+    """Quality of Service enumeration"""
+
     AT_MOST_ONCE = 0
+    """QoS 0 - At most once delivery
+
+    The message is delivered according to the capabilities of the underlying network.
+    No response is sent by the receiver and no retry is performed by the sender.
+    The message arrives at the receiver either once or not at all.
+    """
+
     AT_LEAST_ONCE = 1
+    """QoS 1 - At least once delivery
+
+    This quality of service ensures that the message arrives at the receiver at least once.
+    """
+
     EXACTLY_ONCE = 2
+    """QoS 2 - Exactly once delivery
```

Comment:
Should we note that this is not supported?

Reply:
QoS 2 *is* supported in aws-c-mqtt, but not supported by the AWS IoT Core server. QoS 2 doesn't seem optional from my reading of the spec, but 😬

I'll throw in some cheeky language like "warning: some MQTT servers have been known to not support QoS 2"